pkg/services/publicdashboards/service/service.go

```diff
@@ -350,7 +361,23 @@ func (pd *PublicDashboardServiceImpl) logIsEnabledChanged(existingPubdash *Publi
 	}
 }
 
-// Checks to see if PublicDashboard.Isenabled is true on create or changed on update
+// Filter out dashboards that user does not have read access to
+func (pd *PublicDashboardServiceImpl) filterDashboardsByPermissions(ctx context.Context, u *user.SignedInUser, publicDashboards []PublicDashboardListResponse) ([]PublicDashboardListResponse, error) {
+	for i := len(publicDashboards) - 1; i >= 0; i-- {
+		hasAccess, err := pd.ac.Evaluate(ctx, u, accesscontrol.EvalPermission(dashboards.ActionDashboardsRead, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(publicDashboards[i].DashboardUid)))
+		// If original dashboard does not exist, the public dashboard is an orphan. We want to list it anyway
+		if err != nil && !errors.Is(err, dashboards.ErrDashboardNotFound) {
+			return nil, err
+		}
+
+		if !hasAccess && err == nil {
+			publicDashboards = append(publicDashboards[:i], publicDashboards[i+1:]...)
```

Comment:
I think it could be more readable if we iterate over `publicDashboards`, then append all allowed dashboards to a new result slice. Do we need to modify the input slice?

```
for _, pubdash := range(publicdashboards) {
    var result = []PublicDashboardListResponse
    result = append(result, pubdash)
}
```

Reply:
Agree, changing to a new result slice creation.

Reply:
@evictorero did this change make it in?

Reply:
Yes!